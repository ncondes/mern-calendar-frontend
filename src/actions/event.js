import { fetchWithToken } from '../helpers/fetch';
import { types } from '../types/types';
import { prepareEvents } from '../helpers/prepareEvents';
import Swal from 'sweetalert2';



export const eventStartAddNew = ( event ) => {
    return async ( dispatch, getState ) => {

        const { uid, name } = getState().auth;

        try {

            const res = await fetchWithToken( 'events', event, 'POST' );
            const body = await res.json();
    
            if ( body.ok ) {

                event.id = body.event.id;
                event.user = {
                    _id: uid,
                    name,
                }

                console.log( event );
                dispatch( eventAddNew( event ) );
            }

        } catch ( error ) {
            console.log( error );
        }


    }
}


const eventAddNew = ( event ) => ({
    type: types.eventAddNew,
    payload: event,
});

export const eventSetActive = ( event ) => ({
    type: types.eventSetActive,
    payload: event,
});

export const eventClearActiveEvent = () => ({
    type: types.eventClearActiveEvent,
});


export const eventStartDelete = () => {
    return async ( dispatch, getState ) => {

        const { id } = getState().calendar.activeEvent;

        try {
            
            const res = await fetchWithToken( `events/${ id }`, {}, 'DELETE' );
            const body = await res.json();

            if ( body.ok ) {
                dispatch( eventDeleted() );
            } else {
                Swal.fire('Error', body.msg, 'error');
            }

        } catch ( error ) {
            console.log( error );
        }

    }
}


const eventDeleted = () => ({
    type: types.eventDeleted,
});


export const eventStartLoading = () => {
    return async ( dispatch ) => {

        try {
            
            const res = await fetchWithToken( 'events' );
            const body = await res.json();
            const events = prepareEvents( body.events );

            dispatch( eventLoaded( events ) )

        } catch ( error ) {
            console.log( error );
        }

    }
}

const eventLoaded = ( events ) => ({
    type: types.eventLoaded,
    payload: events,
})



export const eventStartUpdate = ( event ) => {
    return async ( dispatch ) => {

        try {
            
            const res = await fetchWithToken( `events/${ event.id }`, event, 'PUT' );
            const body = await res.json();

            if ( body.ok ) {
                dispatch( eventUpdated( event ) );
            } else {
                Swal.fire('Error', body.msg, 'error');
            }

        } catch ( error ) {
            console.log( error );
        }

    }
}


const eventUpdated = ( event ) => ({
    type: types.eventUpdated,
    payload: event,
});


export const eventLogout = () => ({
    type: types.eventLogout,
})