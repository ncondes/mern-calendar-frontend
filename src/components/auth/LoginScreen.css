
.login__container {
    align-items: center;
    display: flex;
    height: 100vh;
    width: 100vw;
    padding: 1rem;
    justify-content: center;
}

.login__container form {
    background-color: #FFFFFF;
    width: 100%;
    height: auto;
    padding: 4rem 5rem;
    max-width: 35rem;
    box-shadow: 0 5px 8px 0 rgba(0, 0, 0, 0.2), 0 9px 26px 0 rgba(0, 0, 0, 0.19);
}

.login__container form h2 {
    text-align: center;
}

.login__container form a {
    color: #0D6EFD;
}

.login__links {
    display: flex;
    align-items: center;
    justify-content: space-between;
}

@media (max-width: 640px) {
    .login__container form {
        padding: 4rem 1.5rem;
    }
}