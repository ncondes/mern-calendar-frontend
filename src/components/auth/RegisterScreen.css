
.register__container {
    align-items: center;
    display: flex;
    height: 100vh;
    width: 100vw;
    padding: 1rem;
    justify-content: center;
}

.register__container form {
    background-color: #0D6EFD;
    width: 100%;
    height: auto;
    padding: 4rem 5rem;
    max-width: 35rem;
    box-shadow: 0 5px 8px 0 rgba(0, 0, 0, 0.2), 0 9px 26px 0 rgba(0, 0, 0, 0.19);
}

.register__container form h2 {
    color: #FFFFFF;
    text-align: center;
}

.register__container form a {
    color: #FFFFFF;
}

.register__links {
    display: flex;
    align-items: center;
    justify-content: space-between;
}

@media (max-width: 640px) {
    .register__container form {
        padding: 4rem 1.5rem;
    }
}