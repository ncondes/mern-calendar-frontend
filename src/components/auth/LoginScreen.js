import { Link } from 'react-router-dom';
import './LoginScreen.css';

export const LoginScreen = () => {
    return (
        <div className='login__container'>
            <form className=''>
                <h2 className='mb-5'>Login</h2>
                <div className='mb-3'>
                    <input
                        type="email"
                        className='form-control'
                        placeholder='Email'
                    />
                </div>
                <div class='mb-3'>
                    <input
                        type='password'
                        className='form-control'
                        placeholder='Password'
                    />
                </div>
                <div className='login__links'>
                    <button
                        type='submit'
                        className='btn btn-primary'
                    >
                        Login
                    </button>
                    <Link to='/register'>Create a new account</Link>
                </div>
            </form>
        </div>
    )
}
