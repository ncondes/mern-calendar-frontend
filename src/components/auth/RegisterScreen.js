import { Link } from 'react-router-dom';
import './RegisterScreen.css';

export const RegisterScreen = () => {
    return (
        <div className='register__container'>
            <form className=''>
                <h2 className='mb-5'>Sign Up</h2>
                <div className='mb-3'>
                    <input
                        type="text"
                        className='form-control'
                        placeholder='Name'
                    />
                </div>
                <div className='mb-3'>
                    <input
                        type="email"
                        className='form-control'
                        placeholder='Email'
                    />
                </div>
                <div className='mb-3'>
                    <input
                        type="password"
                        className='form-control'
                        placeholder='Password'
                    />
                </div>
                <div class='mb-3'>
                    <input
                        type='password'
                        className='form-control'
                        placeholder='Confirm Password'
                    />
                </div>
                <div className='register__links'>
                    <button
                        type='submit'
                        className='btn btn-light'
                    >
                        Sign up
                    </button>
                    <Link to='/login'>Have an account yet?</Link>
                </div>
            </form>
        </div>
    )
}
