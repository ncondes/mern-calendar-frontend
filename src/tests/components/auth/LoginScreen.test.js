import moduleName from 'module';
import { mount } from 'enzyme'
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';
import { LoginScreen } from '../../../components/auth/LoginScreen';
import { startLogin } from '../../../actions/auth';
import { MemoryRouter } from 'react-router';




// mock a la funcion
jest.mock('../../../actions/auth', () => ({
    startLogin: jest.fn(),
}))

// config store
const middlewares = [ thunk];
const mockStore = configureStore( middlewares );

const initState = {};
let store = mockStore( initState );

// mock store
store.dispatch = jest.fn();

const wrapper = mount(
    <MemoryRouter>
        <Provider store={ store }>
            <LoginScreen />
        </Provider>
    </MemoryRouter>
)



describe('Pruebas en LoginScreen', () => {

    test('Debe mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe de llamar el dispatch del login', () => {
        
        wrapper.find('input[name="loginEmail"]').simulate('change', {
            target: {
                name: 'loginEmail',
                value: 'Juan@gmail.com'
            }
        })
        wrapper.find('input[name="loginPassword"]').simulate('change', {
            target: {
                name: 'loginPassword',
                value: '123456'
            }
        })

        wrapper.find('form').prop('onSubmit')({
            preventDefault(){}
        })

        expect( startLogin ).toHaveBeenCalledWith(  "Juan@gmail.com", "123456" );

    })
    
    
})