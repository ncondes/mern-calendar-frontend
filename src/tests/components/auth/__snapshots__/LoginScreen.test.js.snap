// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en LoginScreen Debe mostrarse correctamente 1`] = `
<div
  className="login__container"
>
  <form
    className=""
    onSubmit={[Function]}
  >
    <h2
      className="mb-5"
    >
      Login
    </h2>
    <div
      className="mb-3"
    >
      <input
        className="form-control"
        name="loginEmail"
        onChange={[Function]}
        placeholder="Email"
        type="email"
        value=""
      />
    </div>
    <div
      className="mb-3"
    >
      <input
        className="form-control"
        name="loginPassword"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
    </div>
    <div
      className="login__links"
    >
      <button
        className="btn btn-primary"
        type="submit"
      >
        Login
      </button>
      <a
        href="/register"
        onClick={[Function]}
      >
        Create a new account
      </a>
    </div>
  </form>
</div>
`;
